src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -488,6 +488,22 @@ memory::ptr primitive_inst::allocate_output(engine& _engine, memory_pool& pool,
     if (total_device_input_mem_size > _engine.get_device_info().max_global_mem_size)
         usm_device_allocatable = false;
 
+    bool memory_reuse_by_user = true;
+    for (auto user : _node.get_users()) {
+        if ((user->get_selected_impl() != nullptr) && (user->get_selected_impl()->can_reuse_memory == false)) {
+            memory_reuse_by_user = false;
+            continue;
+        } else if (user->get_selected_impl() == nullptr) {
+            for (auto sub_user : user->get_users()) {
+                if ((sub_user->get_selected_impl() != nullptr) && (sub_user->get_selected_impl()->can_reuse_memory == false)) {
```

Comment:
What if node-><optimized user> -> <optimized_user> -> <optimized_user> ? 
I think this can be made as a recursive lambda function.

Reply:
like this https://github.com/openvinotoolkit/openvino/blob/263652fb93fd11a40b76ea1dcd96923b93493331/src/plugins/intel_gpu/src/graph/layout_optimizer.cpp#L1706

Reply:
Implemented recursive function. Thanks!